src/v0/destinations/mautic/transform.js

```diff
@@ -154,20 +153,29 @@ const identifyResponseBuilder = async (message, Config, endpoint) => {
 
 const process = async (event) => {
   const { message, destination } = event;
-  const { password, subDomainName, userName } = destination.Config;
-  const endpoint = `${BASE_URL.replace('subDomainName', subDomainName)}`;
+  const { password, subDomainName, domainName, domainMethod } = destination.Config;
   if (!password) {
     throw new ConfigurationError(
       'Invalid password value specified in the destination configuration',
     );
   }
-  if (!subDomainName) {
+  let endpoint;
+  if (!subDomainName && !domainName) {
     throw new ConfigurationError(
-      'Invalid sub-domain value specified in the destination configuration',
+      'Please Provide either subDomain or Domain Name',
     );
   }
-  if (!validateEmail(userName)) {
-    throw new ConfigurationError('Invalid user name provided in the destination configuration');
+  // if both are present we will be taking endpoint after checking the domainMethod selected
+  if (subDomainName && domainName) {
+    if (domainMethod === 'subDomainNameOption') {
+      endpoint = `${BASE_URL.replace('subDomainName', subDomainName)}`;
+    } else {
+      endpoint = `${domainName}/api`;
+    }
+  } else if (subDomainName) {
+    endpoint = `${BASE_URL.replace('subDomainName', subDomainName)}`;
+  } else {
+    endpoint = `${domainName}/api`;
```

Comment:
what about we do this? @ItsSudip @anantjain45823 
```
if(subDomainName && domainMethod ===  'subDomainNameOption') {
    endpoint = `${BASE_URL.replace('subDomainName', subDomainName)}`;
} else if (domainName and domainMethod  === 'domainNameOption')
       endpoint = `${domainName}/api`;
} else {
   throw new ConfigurationError(
      'Please choose correct domain method',
    );
}
```

Reply:
Actually this is done in view of those destination definitions that are already existing. So if `subDomainName` is existing and no `domainName` then we are not looking at `domainMethod` which is the case for the old `destDef`

Reply:
For ideally making it backward compatible we need to assign it as :

`domainMethod = config. domainMethod || 'subDomainNameOption'`. Otherwise the logic seems a bit clusmy to me.
I would like to seek @ItsSudip 's opinion on this

Reply:
Yeah, the logic looks a bit clumsy let's make it clear.

Reply:
i have simplified the login to get endpoint .